packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -928,7 +934,10 @@ class ComboBoxInternal extends React.Component<IComboBoxInternalProps, IComboBox
         .map((item, index) => ({ ...item, index }))
         .filter(
           option =>
-            isNormalOption(option) && !option.disabled && getPreviewText(option).toLocaleLowerCase() === updatedValue,
+            isNormalOption(option) &&
+            !option.disabled &&
+            (this.props.caseSensitive ? getPreviewText(option) : getPreviewText(option).toLocaleLowerCase()) ===
```

Comment:
This pattern of conditionally changing the case is used in a few places. I wonder if it would be better to add something like:

```ts
const sensitize = (text: string, caseSensitive: boolean): string => {
  if (caseSensitive) {
    return text.toLocaleLowerCase();
  }
  return text;
};
```